cmd/podman/create.go

```diff
@@ -555,16 +555,15 @@ func parseCreateOpts(ctx context.Context, c *cli.Context, runtime *libpod.Runtim
 		workDir = data.ContainerConfig.WorkingDir
 	}
 
-	// ENTRYPOINT
-	// User input entrypoint takes priority over image entrypoint
-	entrypoint := c.StringSlice("entrypoint")
-	if len(entrypoint) == 0 && data != nil {
+	var entrypoint []string
+	if c.IsSet("entrypoint") {
+		// ENTRYPOINT
+		if err := json.Unmarshal([]byte(c.String("entrypoint")), &entrypoint); err != nil {
+			return nil, errors.Wrapf(err, "unable to translate entrypint %s", c.String("entrypoint"))
+		}
+	} else {
 		entrypoint = data.ContainerConfig.Entrypoint
 	}
-	// if entrypoint=, we need to clear the entrypoint
-	if len(entrypoint) == 1 && c.IsSet("entrypoint") && strings.Join(c.StringSlice("entrypoint"), "") == "" {
```

Comment:
We need to keep this logic - setting entrypoint explicitly to "" should clear entrypoint